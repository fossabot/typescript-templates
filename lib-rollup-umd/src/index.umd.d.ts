export as namespace LibRollupUmd;
export * from "./index";
