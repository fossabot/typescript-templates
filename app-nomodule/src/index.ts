/// <reference types="@adriangodong/lib-rollup-umd" />

const cls = new LibRollupUmd.SampleClass();
cls.helloWorld();

const subCls = new LibRollupUmd.sub.SampleClass();
subCls.helloWorld();
