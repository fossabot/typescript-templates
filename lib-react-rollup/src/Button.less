@import "./Base.less";

@default-color: #AACCFF;

.default-button {
    color: @default-color
}