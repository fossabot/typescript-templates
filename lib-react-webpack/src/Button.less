@import "./Base.less";

@default-color: #CCFFAA;

:local(.default-button) {
    color: @default-color
}