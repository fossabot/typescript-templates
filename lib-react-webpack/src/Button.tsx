import * as React from "react";

import styles from "./Button.less";

interface ButtonProps
{
    label?: string;
}

export class Button extends React.Component<ButtonProps>
{
    public render(): React.ReactNode
    {
        // const buttonClassName = styles["default-button"];
        const buttonClassName = styles.defaultButton;

        if (buttonClassName === undefined)
        {
            window.console.warn(`Couldn't find buttonClassName from CSS Modules.`, styles);
        }

        return (<button className={buttonClassName}>{this.props.label}</button>);
    }
}