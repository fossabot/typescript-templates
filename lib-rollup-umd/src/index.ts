import * as _sub from "./sub/index";

export * from "./SampleClass";
export * from "./SampleFunc";
export * from "./SampleStaticFunc";
export * from "./UnusedModule";
export const sub = _sub;