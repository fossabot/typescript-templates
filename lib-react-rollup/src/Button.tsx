import * as React from "react";
import "./Button.less";

interface ButtonProps
{
    label?: string;
}

export class Button extends React.Component<ButtonProps>
{
    public render(): React.ReactNode
    {
        return (<button className="default-button">{this.props.label}</button>);
    }
}