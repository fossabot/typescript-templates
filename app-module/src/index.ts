import * as React from "react";
import * as ReactDOM from "react-dom";

import { Button as RollupButton } from "@adriangodong/lib-react-rollup";
import { Button as WebpackButton } from "@adriangodong/lib-react-webpack";
import { SampleClass, sub } from "@adriangodong/lib-rollup-umd";

const cls = new SampleClass();
cls.helloWorld();

const subCls = new sub.SampleClass();
subCls.helloWorld();

ReactDOM.render(
    React.createElement(RollupButton, { label: "Hello world from Rollup!" }),
    document.getElementById("reactRollupRoot")
);

ReactDOM.render(
    React.createElement(WebpackButton, { label: "Hello world from Webpack!" }),
    document.getElementById("reactWebpackRoot")
);