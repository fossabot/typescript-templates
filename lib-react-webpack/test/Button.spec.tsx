import * as React from "react";
import { create } from "react-test-renderer";

import { Button } from "../src/Button";

test("Button", () =>
{
    const tree = create(
        <Button />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test("Button with label", () =>
{
    const tree = create(
        <Button label="label" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});