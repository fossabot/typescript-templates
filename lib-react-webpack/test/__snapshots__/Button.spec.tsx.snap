// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `<button />`;

exports[`Button with label 1`] = `
<button>
  label
</button>
`;
